packages/dataviews/src/dataviews.js

```diff
@@ -28,15 +29,34 @@ export default function DataViews( {
 	actions,
 	data,
 	getItemId = defaultGetItemId,
+	getItemTitle,
 	isLoading = false,
 	paginationInfo,
 	supportedLayouts,
 	onSelectionChange = defaultOnSelectionChange,
 	onDetailsChange = null,
 	deferredRendering = false,
+	labels,
 } ) {
 	const [ selection, setSelection ] = useState( [] );
 
+	useEffect( () => {
+		if (
+			selection.length > 0 &&
+			selection.some(
+				( id ) => ! data.some( ( item ) => item.id === id )
+			)
```

Comment:
What would you think of rewriting this as:

```suggestion
			selection.some( ( id ) => ! data.includes( id ) )
```

Reply:
Ah, data is an object, not a list. We could still do (pseudocode):

```javascript
useEffect( () => {
  const dataIds = data.map( { id } => id );
  if ( selection.some( ( id ) => ! dataIds.includes( id ) ) {
    const newSelection = selection.filter( ( id ) => dataIds.includes( id );
  } 
}, [ selection, data ] );
```

Perhaps it's just my brain today, but I found concatenating two `some` conditions a bit confusing.

Reply:
Fixed 👍

Reply:
Hi @oandregal, I needed to revert this change, data.includes( id ) does not work because data is an array of templates and not an array of ids.

Reply:
Yeah, I provided a second example that works. It's not a blocker, though.